File: 백준/Silver/7795. 먹을 것인가 먹힐 것인가/먹을 것인가 먹힐 것인가.c
#include <stdio.h>
#include <stdlib.h>

// 비교 함수 (qsort에서 사용)
int compare(const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}

int main() {
    int len;
    scanf("%d", &len);

    for (int i = 0; i < len; i++) {
        int rs = 0;
        int sizeA, sizeB;
        
        // 입력 받기
        scanf("%d %d", &sizeA, &sizeB);
        
        int lisA[sizeA], lisB[sizeB];

        for (int j = 0; j < sizeA; j++)
            scanf("%d", &lisA[j]);

        for (int j = 0; j < sizeB; j++)
            scanf("%d", &lisB[j]);

        // B 배열 정렬
        qsort(lisB, sizeB, sizeof(int), compare);

        // A의 각 요소에 대해 B의 요소와 비교
        for (int j = 0; j < sizeA; j++) {
            int num = lisA[j];

            for (int k = 0; k < sizeB; k++) {
                if (num > lisB[k])
                    rs++;
                else
                    break;  // 정렬된 상태이므로 이후는 비교 불필요
            }
        }

        printf("%d\n", rs);
    }

    return 0;
}

File: 백준/Silver/7795. 먹을 것인가 먹힐 것인가/먹을 것인가 먹힐 것인가.cs
int len = int.Parse(Console.ReadLine());

for (int i = 0; i < len; i++)
{
    int rs = 0;
    string[] str = Console.ReadLine().Split();

    int sizeA = int.Parse(str[0]);
    int sizeB = int.Parse(str[1]);

    int[] lisA = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
    int[] lisB = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);

    Array.Sort(lisB);

    foreach (var num in lisA)
    {
        foreach (var value in lisB)
        {
            if (num > value) rs++; 
            else break;
        }
    }

    Console.WriteLine(rs);
}